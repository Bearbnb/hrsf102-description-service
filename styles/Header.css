.container {
  width: 50%;
  font-family: 'Roboto Condensed', sans-serif;
  margin-top: 10px;
  margin-bottom: 10px;
  margin: auto;
}