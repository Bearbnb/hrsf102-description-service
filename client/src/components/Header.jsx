import React from 'react';
import PropTypes from 'prop-types';
import styles from '../../../styles/Header.css';

const Header = ({
  beds, baths, name, ownerName, location, livingSpace,
}) => (
  <div className={styles.container}>
    <div>
      <span className={styles.livingSpace}>ENTIRE {livingSpace.toUpperCase()}</span>
    </div>
    <div className={styles.name}>
      {name}
    </div>
    <div>
      {location}
    </div>
  </div>
);

Header.propTypes = {
  beds: PropTypes.number,
  baths: PropTypes.number,
  ownerName: PropTypes.string,
  name: PropTypes.string,
  location: PropTypes.string,
  livingSpace: PropTypes.string,
};

Header.defaultProps = {
  beds: 0,
  baths: 0,
  ownerName: 'Charlie P',
  name: 'Cozy thing',
  location: 'San Francisco',
  livingSpace: 'Apartment',
};

export default Header;
