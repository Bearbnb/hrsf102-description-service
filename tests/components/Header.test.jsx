import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from '../../client/src/components/Header';

describe('<Header />', () => {
  beforeEach(() => {
    const wrapper = shallow(<Header />);
  });

  it('should render Header properly', () => {
    const wrapper = shallow(<Header />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have props passed to it', () => {
    const wrapper = shallow(<Header />);
    wrapper.setProps({
      beds: 2,
      baths: 3,
      name: 'Ridiculously Overpriced Chateau',
      ownerName: 'John Smith',
      location: 'Brooklyn',
      livingSpace: 'Condo',
    });

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
