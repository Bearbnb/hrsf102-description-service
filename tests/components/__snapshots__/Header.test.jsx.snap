// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render properly 1`] = `
<div
  className="container"
>
  <div>
    <span
      className="livingSpace"
    >
      ENTIRE 
      APARTMENT
    </span>
  </div>
  <div
    className="name"
  >
    Cozy thing
  </div>
  <div>
    San Francisco
  </div>
</div>
`;
